.gitignore

```diff
@@ -1,34 +1,7 @@
-*.o
-*~
-Makefile.in
-aclocal.m4
-autom4te.cache/
-config.guess
-config.h.in
-config.sub
-configure
-compile
-depcomp
-install-sh
-ltmain.sh
-m4/libtool.m4
-m4/ltoptions.m4
-m4/ltsugar.m4
-m4/ltversion.m4
-m4/lt~obsolete.m4
-missing
 nodecachefilereader
 osm2pgsql
 
-Makefile
-config.h
-config.log
-config.nice
-config.status
-libtool
-.deps/
-stamp-h1
-INSTALL
+build
 
 log
 test-driver
```

Comment:
When you run test through CTest the logs and stuff will be stored in Testing/ inside the build directory. Since in-source builds are forbidden and "build" is already ignored I don't think anything special has to be added.